urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/Emotion.java

```diff
@@ -0,0 +1,148 @@
+package weibo4j;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import weibo4j.http.Response;
+import weibo4j.org.json.JSONArray;
+import weibo4j.org.json.JSONException;
+import weibo4j.org.json.JSONObject;
+
+/**
+ * @author SinaWeibo
+ *
+ */
+public class Emotion extends WeiboResponse{
+	private static final long serialVersionUID = -4096813631691846494L;
+
+	private String phrase;
+
+	private String type;
+
+	private String url;
+
+	private boolean is_hot;
+
+	private boolean is_common;
+
+	private int order_number;
+
+	private String category;
+
+	public Emotion(Response res) throws WeiboException {
+		super(res);
+		JSONObject json = res.asJSONObject();
+		try {
+			phrase = json.getString("phrase");
+			type = json.getString("type");
+			url = json.getString("url");
+			is_hot= json.getBoolean("is_hot");
+			order_number = json.getInt("order_number");
+			category = json.getString("category");
+			is_common = json.getBoolean("is_common");
+		} catch (JSONException je) {
+			throw new WeiboException(je.getMessage() + ":" + json.toString(),
+					je);
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 35:46, 49:60.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.